pkg/machines/components/create-vm-dialog/password.jsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cockpit from "cockpit";
+import { FormGroup } from "patternfly-react";
+
+import './password.css';
+
+const _ = cockpit.gettext;
+
+const messages = {
+    weakPassword: _("Password is too weak"),
+    rejectedPassword: _("Password is not acceptable")
+};
+
+export class Password extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            passwordQuality: "",
+            warningQuality: "",
+            displayWarningQual: false,
+            pwscoreAvailable: undefined,
+        };
+        this.handleChangePassword = this.handleChangePassword.bind(this);
+        this.handleWarnings = this.handleWarnings.bind(this);
+        this.validateQuality = this.validateQuality.bind(this);
+        this.passwordQualityFail = this.passwordQualityFail.bind(this);
+        this.passwordQualityPass = this.passwordQualityPass.bind(this);
+    }
+
+    componentDidMount() {
+        cockpit.spawn(["which", "pwscore"], { err: "message" })
+                .then(() => this.setState({ pwscoreAvailable: true }),
+                      () => this.setState({ pwscoreAvailable: false }));
+    }
+
+    handleChangePassword(event) {
```

Comment:
This function is called for each event right now, which incurs an expensive external program  call. I suggest to use throttle-debounce for this, this would also alleviate the need for the setTimeout() in handleWarnings.